### PR TITLE
feat(nx-plugin): initial commit page generator/schematic

### LINKED ISSUE #565: Feature: generate page components
### Which scope/s are relevant/related to the feature request?

nx-plugin

### Information

This feature would add the ability to generate pages. It would support Nx Generators and Angular CLI schematics.

Some examples

ng g page about -> src/app/pages/about.page.ts

ng g page blog/post -> src/app/pages/blog/post.page.ts

ng g page products/[products] -> src/app/pages/products/[products].page.ts

ng g blog -> 
  src/app/content/my-first-post.md
  src/app/pages/blog.page.ts -> blog layout page
  src/app/pages/blog/[slug].page.ts -> blog post page
  src/app/pages/blog/index.page.ts -> blog list page

ng g post my-second-post -> src/content/my-second-post.md

### Describe any alternatives/workarounds you're currently using

_No response_

### I would be willing to submit a PR to fix this issue

- [ ] Yes
- [ ] No

### PR DESCRIPTION
**Pull request by luishcastroc**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/analogjs/analog/blob/main/CONTRIBUTING.md#-commit-message-guidelines
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## Which package are you modifying?

- [ ] vite-plugin-angular
- [ ] vite-plugin-nitro
- [ ] astro-angular
- [ ] create-analog
- [ ] router
- [ ] platform
- [ ] content
- [x] nx-plugin
- [ ] trpc

## What is the current behavior?

There are no generators for Analog plugin

Closes #565 

## What is the new behavior?

This feature would add the ability to generate pages. It would support Nx Generators and Angular CLI schematics.

Some examples

ng g page about -> src/app/pages/about.page.ts

ng g page blog/post -> src/app/pages/blog/post.page.ts

ng g page products/[products] -> src/app/pages/products/[products].page.ts

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: packages/nx-plugin/src/generators/page/schema.json [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+      "x-dropdown": "projects"
+    },
+    "metadata": {
+      "description": "If metadata is needed for the route.",
+      "type": "boolean",
+      "default": false
+    },
+    "title": {
+      "type": "string",
+      "description": "The path to redirect to.",
```
Comment (brandonroberts) [inline packages/nx-plugin/src/generators/page/schema.json:34]:
```suggestion
      "description": "The page title.",
```

File: packages/nx-plugin/src/generators/page/schema.json [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    },
+    "redirectPage": {
+      "description": "Whether the generated page is a redirect. _Note: Redirect path needs to be provided in the next prompt._",
+      "type": "boolean",
+      "default": false,
+      "x-priority": "important",
+      "x-prompt": "Is this a redirect page?"
+    },
+    "redirectPath": {
+      "description": "The path to redirect to. _Note: This is only needed if the previous prompt is set to true._",
```
Comment (brandonroberts) [inline packages/nx-plugin/src/generators/page/schema.json:45]:
```suggestion
      "description": "The path to redirect to. _Note: This is only needed if the redirect page is set to true._",
```

File: pnpm-lock.yaml [hunk 3]
```diff
@@ -5379,7 +5379,7 @@ packages:
     dependencies:
       '@babel/core': 7.22.5
       '@jest/types': 29.5.0
-      '@jridgewell/trace-mapping': 0.3.17
```
Comment (brandonroberts) [inline pnpm-lock.yaml:5382]: Revert the changes to this file

File: packages/nx-plugin/src/generators/page/schema.json [hunk 4]
```diff
@@ -0,0 +1,60 @@
+{
+  "$schema": "http://json-schema.org/schema",
+  "cli": "nx",
+  "$id": "page",
+  "title": "Create an Analog page",
+  "type": "object",
+  "properties": {
+    "name": {
```
Comment (brandonroberts) [inline packages/nx-plugin/src/generators/page/schema.json:8]: I think this should be `path` or `pathname`. It represents the URL path to the page.

**Review by brandonroberts — CHANGES_REQUESTED** [review]

**Review by brandonroberts — APPROVED** [review]

**Comment by luishcastroc** [comment]
should i complete this one with the blog and post generators? or should we merge it like this and continue in a different complementary feature @brandonroberts

**Comment by brandonroberts** [comment]
@luishcastroc let's merge this as-is and do the blog support separately. Also need to add some docs for using the generators/schematics

**Comment by luishcastroc** [comment]
Got it let me work on that and finish those with documentation so we can have everything in place

**Pull request merged**